en/03_Upgrading/07_Deprecations.md

```diff
@@ -4,27 +4,108 @@ title: Deprecations
 
 Major releases of Silverstripe CMS introduce many API changes. They include a deprecation message that will either tell you:
 - to simply use a different yet equivalent API instead, or
 - the API in question has no replacement. This is usually for API that was mostly irrelevant.
 
-To see deprecation warnings in your project and in your project's CI, add `SS_DEPRECATION_ENABLED=true` to your project's `.env` file.
+## Enabling deprecation warnings
+
+To enable deprecation warnings, set the `SS_DEPRECATION_ENABLED` environment variable in your project's `.env` file.
+
+[info]
+If the `SS_DEPRECATION_ENABLED` environment variable is set, this takes precedence over use of the `Deprecation::enable()` static method.
+[info]
+
+```env
+SS_DEPRECATION_ENABLED=true
+```
+
+Alternatively, add the following line to your project's `app/_config.php`.
 
-Alternatively, add the following line to you project's `app/_config.php`.
 ```php
-Deprecation::enable()
+Deprecation::enable();
+```
+
+[info]
+Deprecation warnings will only ever show if your `SS_ENVIRONMENT_TYPE` is set to `dev`.
+[/info]
+
+Once you have resolved all of the deprecation warnings you can, it is recommended to turn off deprecation warnings again.
+
+Not all API that gets deprecated will have an equivalent replacement API in that same major version; some of the API is only available from the next major release. A good example of this is the upgrade for what powers the `SilverStripe\Control\Email\Email` class from `swiftmailer` in CMS 4 to `symfony/mailer` in CMS 5. In these cases, you'll need to upgrade to the new major version before you can access the replacement API.
+
+Some code that has been deprecated with no immediate replacement will not emit deprecation notices by default. If you wish to also see notices for deprecated code with no immediate replacement, add the following line to you project's `app/_config.php` file. Note that this will also emit deprecation notices for usages of the deprecated code inside core modules.
+
+```php
+Deprecation::enable(true);
+```
+
+## How to view deprecation warnings
+
+By default, deprecation warnings will be emitted to the error logger, and will be output at the end of CLI responses. They will not be included in HTTP responses by default.
+
+### Viewing deprecation warnings in the logs
+
+Deprecation warnings are output to the same error logger as all other warnings and errors. You will need to make sure you have a logging handler attached to the default `Psr\Log\LoggerInterface` or `Psr\Log\LoggerInterface.errorhandler` singletons. For example, to log to a file you can add this to your yml configuration:
+
+```yml
+SilverStripe\Core\Injector\Injector:
+  ErrorLogFileHandler:
+    class: Monolog\Handler\StreamHandler
+    constructor:
+      - "../silverstripe.log" # an absolute path, or a path relative to the index.php file (usually inside the public/ directory)
```

Comment:
Permissions are the responsibility of the developer.

Reply:
Yes permissions are the developers responsibility, however these instructions are just flat out bad?

If I copy paste the above with `../silverstripe.log` into `app/_config/mysite.yml` and dev/build flush=1 - I get a PHP Fatal error

It's trying to create the log file in a directory ABOVE where my project is.  The code comment `or a path relative to the index.php file (usually inside the public/ directory)` is just wrong? It's trying to create it relative to the project root.  Seems like the intention was for it to create the log file in the project root, but somewhere along the lines something changed

If I instead use `./silverstripe.log` things work fine

Reply:
> I get a PHP Fatal error

Because you haven't set the correct permissions.
I'll add a note about permissions in a `[notice]` block. But adding the log to `public/silverstripe.log` is an objectively bad instruction, so I'm not going to do that.

Reply:
> The code comment or a path relative to the index.php file (usually inside the public/ directory) is just wrong? It's trying to create it relative to the project root.

I'll validate that - if it's correct, I'll update the docs in both places about this and use the correct path.

Reply:
I've just tested this. `./silverstripe.log` does indeed use the file in `public/silverstripe.log`, so the current instructions are correct.

Reply:
done

Reply:
Replicated issue locally

Setup
- Copy paste yml instructions above to app/_config/mylogging.yml
- Construct a deprecated class in PageController::init() to trigger error
- Call Deprecation::enable() in _config.php

When running dev/build flush=1 from CLI:
- ./silverstripe.log will be created in root dir
- ../silverstripe.log will say permission error, trys to create /var/www/../silverstripe.log

When requesting page via browser
- ./silverstripe.log will be created in public dir
- ../silverstripe.log will be created in root dir

Reply:
I've added acceptance criteria to https://github.com/silverstripe/silverstripe-framework/issues/10551 and bumped its impact level. Whatever docs changes are needed re: file paths for logging will be done as a part of that issue.